hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/ActionRepository.java

```diff
@@ -352,6 +352,22 @@ Long countByRolloutAndRolloutGroupAndStatusNotAndStatusNotAndStatusNot(JpaRollou
             JpaRolloutGroup rolloutGroupParent, Status actionStatus);
 
     /**
+     * Retrieving all actions referring to a given rollout group with a specific
+     * a specific status.
+     *
+     * @param rollout
+     *            the rollout the actions belong to
+     * @param rolloutGroupParent
+     *            the parent rollout group the actions should reference
+     * @param actionStatus
+     *            the status the actions have
+     * @return the actions referring a specific rollout group and a specific
+     *         status
+     */
+    List<Identifiable<Long>> findByRolloutAndRolloutGroupAndStatus(JpaRollout rollout,
```

Comment:
In the List are `Identifiable` and not `Long` objects. So i think the list contains `Action` objects  because a `Action` is an instance of an `Identifiable`. So i don't see any memory optimization in the java code. 
Maybe spring data loads less information to the `Action` entity (only the id).